enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/lucene/FulltextFactory.java

```diff
@@ -76,21 +72,14 @@ private Analyzer getAnalyzer( String analyzerClassName )
         return analyzer;
     }
 
-    LuceneFulltext createFulltextIndex( String identifier, FulltextIndexType type, List<String> properties )
+    public LuceneFulltext createFulltextIndex( long indexId, FulltextIndexDescriptor descriptor )
     {
-        File indexRootFolder = new File( indexDir, identifier );
-        LuceneIndexStorageBuilder storageBuilder = LuceneIndexStorageBuilder.create();
-        storageBuilder.withFileSystem( fileSystem ).withIndexFolder( indexRootFolder );
-        PartitionedIndexStorage storage = storageBuilder.build();
-        return new LuceneFulltext( storage, partitionFactory, properties, analyzer, identifier, type );
+        PartitionedIndexStorage storage = indexStorageFactory.indexStorageOf( indexId, config.get( GraphDatabaseSettings.archive_failed_index ) );
```

Comment:
`indexStorageFactory.indexStorageOf( indexId, config.get( GraphDatabaseSettings.archive_failed_index ) )` should be extracted into separte method